cmd/agent/clcrunnerapi/clc_runner_server.go

```diff
@@ -0,0 +1,113 @@
+// Unless explicitly stated otherwise all files in this repository are licensed
+// under the Apache License Version 2.0.
+// This product includes software developed at Datadog (https://www.datadoghq.com/).
+// Copyright 2016-2019 Datadog, Inc.
+
+/*
+Package clcrunnerapi implements the clc runner IPC api. Using HTTP
+calls, the cluster Agent collects stats to optimize the cluster level checks dispatching.
+*/
+package clcrunnerapi
+
+import (
+	"crypto/tls"
+	"crypto/x509"
+	"encoding/pem"
+	"fmt"
+	stdLog "log"
+	"net"
+	"net/http"
+	"time"
+
+	v1 "github.com/DataDog/datadog-agent/cmd/agent/clcrunnerapi/v1"
+	"github.com/DataDog/datadog-agent/pkg/api/security"
+	"github.com/DataDog/datadog-agent/pkg/api/util"
+	"github.com/DataDog/datadog-agent/pkg/config"
+	"github.com/gorilla/mux"
+)
+
+var (
+	clcListener net.Listener
+)
+
+// StartCLCRunnerServer creates the router and starts the HTTP server
+func StartCLCRunnerServer() error {
+	// create the root HTTP router
+	r := mux.NewRouter()
+
+	// IPC REST API server
+	v1.SetupHandlers(r.PathPrefix("/api/v1").Subrouter())
+
+	// Validate token for every request
+	r.Use(validateCLCRunnerToken)
```

Comment:
Maybe this is something for you guys to follow up with the security guys.... how are these tokens distributed to the remote agents? I guess this is responsibility of the user, do we have some best practices for them?

Reply:
we recommend using a kubernetes secret to distribute tokens (this will be documented for sure), the same way we recommend it to set up the cluster agent https://docs.datadoghq.com/agent/kubernetes/cluster/#secure-cluster-agent-to-agent-communication